### ISSUE TITLE
Getting Fatal Error: Class Crypt_RSA not found

### ISSUE THREAD

**Issue by redjersey** [Petertjuh360/da-letsencrypt#64]
I have installed da-letsencrypt, with php = 5.5.x and have sni enabled.

In DA, when I go to the user level, select the domain / subdomain and tried to request a SSL certificate, click on the TOS checkbox and submit, I got:

Fatal error: Class 'Crypt_RSA' not found in /usr/local/directadmin/plugins/da-letsencrypt/includes/lib/Account.php on line 94

**Comment by Wouter0100** [comment]
Hmm. It seems like you still have problems with your dependencies. `Crypt_RSA` is one of them. What do you get when you do a `composer update` in the plugin dir? (please past it within ticks or through github gist)

**Comment by redjersey** [comment]
this is what I got:

git pull
git: /usr/local/lib/libz.so.1: no version information available (required by git)
git: /usr/local/lib/libz.so.1: no version information available (required by git)
git: /usr/local/lib/libz.so.1: no version information available (required by git)
git: /usr/local/lib/libz.so.1: no version information available (required by git)
git: /usr/local/lib/libz.so.1: no version information available (required by git)
git: /usr/local/lib/libz.so.1: no version information available (required by git)
git: /usr/local/lib/libz.so.1: no version information available (required by git)
git: /usr/local/lib/libz.so.1: no version information available (required by git)
git-remote-https: /usr/local/lib/libz.so.1: no version information available (required by git-remote-https)
git: /usr/local/lib/libz.so.1: no version information available (required by git)
remote: Counting objects: 4, done.
remote: Compressing objects: 100% (4/4), done.
remote: Total 4 (delta 0), reused 0 (delta 0), pack-reused 0
git: /usr/local/lib/libz.so.1: no version information available (required by git)
Unpacking objects: 100% (4/4), done.
From https://github.com/Petertjuh360/da-letsencrypt
   ba896c6..7e58182  master     -> origin/master
git: /usr/local/lib/libz.so.1: no version information available (required by git)
git: /usr/local/lib/libz.so.1: no version information available (required by git)
git-merge: /usr/local/lib/libz.so.1: no version information available (required by git-merge)
Updating ba896c6..7e58182
error: Your local changes to 'scripts/cron.php' would be overwritten by merge.  Aborting.
Please, commit your changes or stash them before you can merge.

# php composer.phar update

Loading composer repositories with package information
Updating dependencies (including require-dev)
- Installing sabre/uri (1.0.1)
  Loading from cache

and I can't still get rid of the Fatal error: Class 'Crypt_RSA' not found error

**Comment by barryvdh** [comment]
@Wouter0100 Are you sure that Crypt_RSA is a dependancy? It's a PEAR repo, which migrated to phpseclib (https://github.com/phpseclib/phpseclib)
Should we require `phpseclib/phpseclib:2.x` and use the full namespace `phpseclib\Crypt\RSA` instead of Crypt_RSA?

**Comment by barryvdh** [comment]
See https://pear.php.net/package/Crypt_RSA

> This package is not maintained anymore and has been superseded. Package has moved to channel phpseclib.sourceforge.net, package Crypt_RSA.
> **Summary**
> Provides RSA-like key generation, encryption/decryption, signing and signature checking. Users are strongly advised to migrate to phpseclib's Crypt_RSA; which is better maintained and less vulnerable to security issues.

Perhaps the PEAR package is installed on some servers by default.

**Comment by Wouter0100** [comment]
Hmm, yeah. We should add that one, I guess we should receive it through another dependency, but we're also using this one.

**Comment by zenire** [comment]
The comment of @barryvdh to use `phpseclib\Crypt\RSA is tested by an user of the plugin and it works fine.

**Comment by Wouter0100** [comment]
Hmm, if that user could open a PR, would be awesome ;)

**Comment by WaaaghNL** [comment]
Just messed up a install of this plugin. Removed everything and pulled the plugin again. Now i got this error also.

**Comment by Wouter0100** [comment]
Hmm, can you add the package to the `composer.json` and test again?

**Comment by WaaaghNL** [comment]
Like so? 
`{
  "require": {
    "kelunik/acme": "^0.2.1",
    "php": ">=5.5.0",
    "phpseclib/phpseclib":"2.x"
  },
  "minimum-stability": "dev",
  "autoload": {
    "psr-4": {
      "DirectAdmin\\LetsEncrypt\\Lib\\": "includes/lib"
    }
  }
}`

and then run `cd /usr/local/directadmin/plugins/da-letsencrypt
git pull
composer update`

Output: 
`[root@web20 da-letsencrypt]# composer update
Loading composer repositories with package information
Updating dependencies (including require-dev)
Nothing to install or update
Writing lock file
Generating autoload files`

Output in DA:
`Fatal error: Class 'Crypt_RSA' not found in /usr/local/directadmin/plugins/da-letsencrypt/includes/lib/Account.php on line 94`

**Comment by barryvdh** [comment]
Thats a different class name.

**Comment by WaaaghNL** [comment]
Uuhm what do i need to correct. I think i'm missing someting ;-)

**Comment by stijnvanouplines** [comment]
I'm getting the same error with a fresh install of the plugin.

**Comment by zenire** [comment]
@svouplin Which version of PHP are you using?

**Comment by stijnvanouplines** [comment]
@Petertjuh360 I'm using PHP 5.5.31

**Comment by Swissgroove** [comment]
Had the same error. Went to Admin level -> "Let's Encrypt Certificates" -> selected "Let's Encrypt Live" and saved. Then back to user level to request SSL certificate for domain and it worked.

**Comment by stijnvanouplines** [comment]
When I go to Admin level -> "Let's Encrypt Certificates" I get an error:
"Please select a domain first at the DirectAdmin homepage."

**Comment by Wouter0100** [comment]
@svouplin please create another issue for this with debug information.

@Swissgroove That won't solve the problem for sure.

**Comment by stijnvanouplines** [comment]
@Wouter0100 That error is now gone. And the solution from @Swissgroove doesn't work for me.

**Comment by redjersey** [comment]
why this is close? how to fix that problem? even if I go to admin letsencrypt and set it to live, I'm still getting the Class Crypt_RSA not found  error

**Comment by Wouter0100** [comment]
Because there is an open PR which fixes this.

**Comment by WaaaghNL** [comment]
But the problem isn't fixt with the PR, still the same error on the user page. pulled 5min ago as clean install...

**Comment by ghost** [comment]
@WaaaghNL The pull request has to be merged first.

**Comment by WaaaghNL** [comment]
oops my bad

**Comment by Wouter0100** [comment]
It doesn't need to be merged. If you switch to the `da-letsencrypt-64`-branch, it should be working. If you're able to confirm it's working properly, that it's ready to be merged in master :)

**Comment by WaaaghNL** [comment]
@Wouter0100 it's working, but i'm back to #57

**Comment by stijnvanouplines** [comment]
<code>da-letsencrypt-64</code> branch is working fine!

**Comment by Wouter0100** [comment]
It's already merged in `master`.
